### ISSUE TITLE
smali gradle plugin?

### ISSUE THREAD

**Issue by andrewleech** [JesusFreke/smali#662]
Just wondering if you've looked at / considered wrapping smali up as a gradle compilation plugin to allow mixing smali source alongside java source in an Android / gradle project?

https://guides.gradle.org/designing-gradle-plugins/
https://docs.gradle.org/current/userguide/standard_plugins.html#sec:language_plugins

I'd like to have a go at doing this myself, I've never looked into gradle internals so I really don't know how much effort it would take.

Thanks for an amazing tool either way!

**Comment by JesusFreke** [comment]
Yeah, I've considered something like that, but haven't pursued it. It should definitely be do-able, but you would probably need to generate java stubs from the smali classes, in order to get the java compilation to work. It would definitely be an interesting project, but not something I'm planning on doing any time soon. But if you want a project... :D

**Comment by GemiDroid** [comment]
> Yeah, I've considered something like that, but haven't pursued it. It should definitely be do-able, but you would probably need to generate java stubs from the smali classes, in order to get the java compilation to work. It would definitely be an interesting project, but not something I'm planning on doing any time soon. But if you want a project... :D

Could you help me ?
I converted code to .smali and need to convert it to java code to recompile it, how can I do that ?

**Comment by andrewleech** [comment]
@Gemdroid that's something not covered by this project, there's no direct conversion from smali to java available. 

There are a number of other java decompilers around that go from apk/jar to java, all with varying levels of success. Most code from a decompiler needs a fair bit of cleaning up to be able to compile again. 
I suggest you try out this project, it's a useful tool that lets you try a bunch of different decompilers on your code: https://github.com/Konloch/bytecode-viewer
